Source/WebKit/NetworkProcess/NetworkProcess.cpp

```diff
@@ -1327,13 +1327,19 @@ void NetworkProcess::setAppBoundDomainsForResourceLoadStatistics(PAL::SessionID
 #endif
 
 #if ENABLE(MANAGED_DOMAINS)
-void NetworkProcess::setManagedDomainsForResourceLoadStatistics(PAL::SessionID sessionID, HashSet<WebCore::RegistrableDomain>&& appBoundDomains, CompletionHandler<void()>&& completionHandler)
+void NetworkProcess::setManagedDomainsForResourceLoadStatistics(PAL::SessionID sessionID, HashSet<WebCore::RegistrableDomain>&& managedDomains, CompletionHandler<void()>&& completionHandler)
 {
     if (auto* session = networkSession(sessionID)) {
-        if (auto* resourceLoadStatistics = session->resourceLoadStatistics()) {
-            resourceLoadStatistics->setManagedDomains(WTFMove(appBoundDomains), WTFMove(completionHandler));
-            return;
-        }
+        supplement<WebCookieManager>()->setHTTPCookieAcceptPolicy(sessionID, WebCore::HTTPCookieAcceptPolicy::AlwaysAccept, [session = WeakPtr { *session }, managedDomains = WTFMove(managedDomains), completionHandler = WTFMove(completionHandler)]() mutable {
+            if (session) {
+                if (auto* resourceLoadStatistics = session->resourceLoadStatistics()) {
```

Comment:
Without making this too complicated, you might want to think about the correct behavior if this condition fails - should we still allow all cookies in this case?

Reply:
That's an interesting point. If we have reached this point, the MDM administrator told WebKit it wanted to operate in this special mode, but for some reason the `NetworkSession` was killed between the time we told CFNetwork about it, and the time we went to update the ITP infrastructure about the domains.

If the `NetworkSession` associated with the `sessionID` is dead, we can't invoke `setHTTPCookieAcceptPolicy` again to change it, because we have no `sessionID` to give it. But that's probably okay, since the cookie state is set on the `CFHTTPCookieStorageRef` associated with `sessionID`. If the session no longer exists, the cookie store should also be gone (at least for Cocoa-based ports).

So I think we either want to leave the code as-is (since the re-spawned NetworkSession will pass through this code again and set the HTTP Cookie Accept policy to the same way, so there would be no behavior change), OR we should RELEASE_ASSERT that session is not `nullptr` and just crash so the UIProcess respawns if a NetworkSession